app/assets/javascripts/ship.js

```diff
@@ -0,0 +1,35 @@
+$(function () {
+
+
+  $('.square').droppable({
+  tolerance: 'intersect',
+  snap: '.square',
+  drop: function(event, ui) {
+    var drop_p = $(this).offset();
+    var drag_p = ui.draggable.offset();
+    var left_end = drop_p.left - drag_p.left + 1;
+    var top_end = drop_p.top - drag_p.top + 1;
+    ui.draggable.animate({
+      top: '+=' + top_end,
+      left: '+=' + left_end
+    });

```

Comment:
In upcoming work, I'd suggest trying to do an AJAX request here to move the piece.